src/automata/FireAndForgetRequester.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <memory>
+
+#include "src/Payload.h"
+
+#include "src/ReactiveStreamsCompat.h"
+#include "src/mixins/ConsumerMixin.h"
+#include "src/mixins/ExecutorMixin.h"
+#include "src/mixins/LoggingMixin.h"
+#include "src/mixins/MemoryMixin.h"
+#include "src/mixins/MixinTerminator.h"
+#include "src/mixins/PublisherMixin.h"
+#include "src/mixins/SinkIfMixin.h"
+#include "src/mixins/SourceIfMixin.h"
+#include "src/mixins/StreamIfMixin.h"
+
+namespace reactivesocket {
+
+/// Implementation of an automaton that represents a Fire-and-Forget requester.
+/// Note that it's simpler than other automata due to the inherent throwaway
+/// simplicity of Fire-and-Forget semantics. Stream IDs are not tracked.
+///
+/// TODO: we need to add logging similar to other *Requester automata.
+class FireAndForgetRequester: public Subscription {
```

Comment:
It's not intuitive to me either why we need to go through the subscriptions/request(N) in fire-and-forget. I'm going by what the Java implementation does: https://github.com/ReactiveSocket/reactivesocket-java/blob/master/reactivesocket-core/src/main/java/io/reactivesocket/internal/Requester.java#L175. (there's also an example in tests https://github.com/ReactiveSocket/reactivesocket-java/blob/master/reactivesocket-core/src/test/java/io/reactivesocket/ReactiveSocketTest.java#L416).